### PR TITLE
Moving Data Analysis GUI interface files to inelastic folder.

### LINKED ISSUE #35665: Move Data analysis GUI
In the indirect GUI's there is a "data analysis" GUI. However, it is also used for direct experiments. This issue is to move it to the inelastic section (including files and docs updated).

### LINKED ISSUE #36323: Move Data Manipulation GUI files
**Describe the outcome that is desired.**
We would like to move the files used for the Data Manipulation GUI to the `qt/scientific_interfaces/Inelastic/Manipulation` folder location. This includes moving the test files for this interface to a `qt/scientific_interfaces/Inelastic/Mantipulation/test` location. 

The desired structure for the new Inelastic category is

```
qt/scientific_interfaces/Inelastic
    -> Analysis/
        *.cpp
        *.h
        -> test/
    -> Manipulation/
        *.cpp
        *.h
        -> test/
```

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
A new qt library / test library for the inelastic interface category has been created.
 A new inelastic subdirectory has been created,  and the data analysis interface files and test files have been moved 
to the new category.
#### Summary of work

Fixes #35665 
Fixes #36323

### To test:

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by robertapplin — CHANGES_REQUESTED** [review]
Code changes look good and the hard part (creating a qt library) is done!

I've just realised how many Data Analysis files there are. We also need to moves these and their related test files:

- [ ] `IFQFitObserver`
- [ ] `IndirectDockWidgetArea`
- [ ] `IndirectFitPlotModel`
- [ ] `IndirectFitPlotView`
- [ ] `IndirectFitPropertyBrowser`
- [ ] `ConvFitModel`
- [ ] `FQFitConstants`
- [ ] `FitStatusWidget`
- [ ] `FunctionTemplateBrowser`
- [ ] `IDAFunctionParameterEstimation`
- [ ] `IIndirectFittingModel`
- [ ] `IIndirectFitDataView`
- [ ] `IIndirectFitOutput`
- [ ] `IIndirectFitOutputOptionsModel`
- [ ] `IIndirectFitOutputOptionsView`
- [ ] `IIndirectFitPlotView`
- [ ] `IIndirectFitPlotModel`
- [ ] `IIndirectFitPlotPresenter`
- [ ] `IIndirectFittingModel`
- [ ] `IndirectEditResultsDialog`
- [ ] `IndirectFitData`
- [ ] `IndirectFitDataModel`
- [ ] `IndirectFitDataPresenter`
- [ ] `IndirectFitDataView`
- [ ] `IndirectFitOutput`
- [ ] `IndirectFitOutputOptionsPresenter`
- [ ] `IndirectFitTab`
- [ ] `IndirectSpectrumSelector`
- [ ] `ParameterEstimation`

**Comment by adriazalvarez** [comment]
I have added the remaining files in the last commit. 
Sorry for the huge file dump. It was very entangled with data analysis and data manipulation, 
so I have decided to merge this issue with #36323 . This last commit makes part of that. 
(at the end there were data manipulation classes calling data analysis classes). Additionally, I had to move the indirect function browser folder to inelastic folder to resolve dependencies in data analysis.
Finally I also moved the test.

File: qt/scientific_interfaces/Indirect/CMakeLists.txt [hunk 1]
```diff
@@ -226,38 +103,24 @@ set(UI_FILES
...
     IndirectAddWorkspaceDialog.ui
     IndirectBayes.ui
     IndirectCorrections.ui
-    IndirectDataAnalysis.ui
     IndirectDataReduction.ui
     IndirectDiffractionReduction.ui
-    IndirectEditResultsDialog.ui
-    IndirectFitDataView.ui
-    IndirectFitOutputOptions.ui
     IndirectFitPreviewPlot.ui
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/CMakeLists.txt:116]: I think we want to move this to the Inelastic/Analysis folder

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h [hunk 2]
```diff
@@ -24,7 +24,7 @@ namespace CustomInterfaces {
 namespace IDA {
 
 // Used for painting an Icon onto the handle of the splitter on workbench
-class SplitterHandle : public QSplitterHandle {
+class MANTIDQT_INELASTIC_DLL SplitterHandle : public QSplitterHandle {
```
Comment (robertapplin) [inline qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h:27]: I think we probably don't need the export macro on this class

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h [hunk 3]
```diff
@@ -41,7 +41,7 @@ class SplitterHandle : public QSplitterHandle {
   QIcon m_icon;
 };
 
-class Splitter : public QSplitter {
+class MANTIDQT_INELASTIC_DLL Splitter : public QSplitter {
```
Comment (robertapplin) [inline qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPlotView.h:44]: And same with this class?

File: qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPropertyBrowser.h [hunk 4]
```diff
@@ -38,10 +38,10 @@ namespace IDA {
 using namespace Mantid::API;
 using namespace MantidWidgets;
 
-class FunctionTemplateBrowser;
-class FitStatusWidget;
+class MANTIDQT_INELASTIC_DLL FunctionTemplateBrowser;
+class MANTIDQT_INELASTIC_DLL FitStatusWidget;
```
Comment (robertapplin) [inline qt/scientific_interfaces/Inelastic/Analysis/IndirectFitPropertyBrowser.h:42]: These are just forward declarations and so will not need the export macro
https://stackoverflow.com/questions/4757565/what-are-forward-declarations-in-c

**Review by robertapplin — CHANGES_REQUESTED** [review]
This new file organisation is much better, thanks! Can you rename the `IndirectFunctionBrowser` folder to `FunctionBrowser`, and move it into the Analysis folder?

File: qt/scientific_interfaces/Indirect/IndirectFitDataModel.h [hunk 5]
```diff
@@ -23,7 +23,7 @@ using namespace MantidWidgets;
    IndirectFitDataModel - Specifies an interface for updating, querying and
    accessing the raw data in IndirectDataAnalysisTabs
 */
```
Comment (robertapplin) [inline qt/scientific_interfaces/Indirect/IndirectFitDataModel.h:25]: I think you might have missed removing this file from the Indirect folder? It is already in the Inelastic/Analysis folder

**Review by robertapplin — APPROVED** [review]
Thanks for all your hard work on this. This is a really good first step for organising the files in Indirect/Inelastic. It lowers the entry threshold for new starters, and also should make it easier for us to spot commonly used functionality that can be shared across interfaces in the future

**Pull request merged**